### ISSUE TITLE
Non linear Protein Binding Function

### ISSUE THREAD

**Issue by hardik1chandasana** [Open-Systems-Pharmacology/Forum#157]
Hello Everyone,

I was wondering if anyone has developed a model with incorporating nonlinear protein binding (Typical and Atypical) function in PK-Sim?

**Comment by prvmalik** [comment]
I believe you can do this by setting fraction unbound to 1 in your compound description and instead doing the protein binding manually.  Add a "protein binding partner," specify the kon and koff binding rate constants and have it present in plasma and tissue at saturable concentrations.

**Comment by msevestre** [comment]
@prvmalik What about partition coefficients? I fear that their calculation is dependent on fu value.

**Comment by StephanSchaller** [comment]
>  What about partition coefficients? I fear that their calculation is dependent on fu value.

Yes, this is especially relevant if you have a small molecule with intracellular exposure. 

For proteins, solution as proposed by @prvmalik might almost work, except that if you define an explicit  protein binding partner you also need to consider that protein binding partners defined this way do not distribute (as albumin does).

But exporting to MoBi and defining fu as a formula dependent parameter instead as a constant may work... Maybe this could be a feature request for PK-Sim?

**Comment by hardik1chandasana** [comment]
@StephanSchaller I will remind you. Thanks

**Comment by hardik1chandasana** [comment]
@StephanSchaller  Gentle reminder about non-linear PPB function.

**Comment by hardik1chandasana** [comment]
@StephanSchaller Thanks for the explanation. I will try and ask you if any help needed.

**Comment by msevestre** [comment]
@hardik1chandasana Please let us know if you have any more questions. Cheers

**Comment by BB82627** [comment]
<img width="713" height="558" alt="Image" src="https://github.com/user-attachments/assets/021c4fde-89e5-4a84-b5f9-f9a97c225f4c" />

<img width="1518" height="191" alt="Image" src="https://github.com/user-attachments/assets/5e9f58ce-359c-4bad-8f9e-3bd6d2f1a4ee" />

i try to set the fup as suggested but why it pops up with notification with this ?

**Comment by PavelBal** [comment]
@BB82627 most probably, you have a formula that is empty (maybe you created it but did not use it later). Switch to the "Formulas" tab in the MOlecules Building Block, and find a formula that has an empty string. Delete the formula, and it should work again.
